internal/definition/validate.go

```diff
@@ -93,3 +98,31 @@ func detectFiles(ctx context.Context, fpath string) ([]string, error) {
 
 	return defFiles, nil
 }
+
+// detect if the string is json
+func detectJson(msg string) bool {
+	var jsMsg json.RawMessage
+	js := json.Unmarshal([]byte(msg), &jsMsg)
+	return js == nil
+}
+
+// create a file in a temp dir with contents of msg
+func stringToFile(ctx context.Context, msg string) ([]string, error) {
+	fs := utils.FS(ctx)
+	tmpDir, err := afero.TempDir(fs, "", "ec-definition")
+	if err != nil {
+		return nil, err
+	}
+
+	fileName, err := utils.GenerateRandomString(10)
+	if err != nil {
+		return nil, err
+	}
+
+	filePath := filepath.Join(tmpDir, fileName)
+	err = afero.WriteFile(fs, filePath, []byte(msg), 0644)
+	if err != nil {
+		return nil, err
+	}
+	return []string{filePath}, nil
+}
```

Comment:
Perhaps something like:
```suggestion
func writeTempFile(ctx context.Context, data string) (string, error) {
	fs := utils.FS(ctx)
	file, err := fs.TempFile("", "*")
	if err != nil {
		return "", err
	}
	path := file.Name()
	if _, err := file.WriteString(data); err != nil {
		_ := fs.Remove(path)
		return "", err
	}
	return path, nil
}
```